### ISSUE TITLE
sequelize.sync() doesn't create all columns

### ISSUE THREAD

**Issue by Branden97** [sequelize/sequelize#14307]
## Issue Creation Checklist

- [x] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

## Bug Description
sequelize.sync() doesn't create all columns if the column definition is a variable used for multiple columns.

Example:
```ts
import { DataTypes, Sequelize } from '@sequelize/core'

const notNullStringColumn = {
    type: DataTypes.STRING,
    allowNull: false,
}

const sequelize = new Sequelize('sqlite::memory:')

const User = sequelize.define('User', {
    firstName: notNullStringColumn,
    lastName: notNullStringColumn,  // <-- this column won't be created!
});

await sequelize.sync()
```

PS: This is my first time opening a github issue - any feedback is welcome :)

### SSCCE

**Here is the link to the SSCCE for this issue:** [LINK-HERE](https://github.com/sequelize/sequelize-sscce/pull/226) 

### What do you expect to happen?

All table columns should have been created by sequelize.sync().

### What is actually happening?

Only the first column defined using the column definition variable (per table) is created (see SSCCE).

### Environment

- Sequelize version: @sequelize/core@7.0.0-alpha.10 (also sequelize@6.17.0 in the SSCCE)
- Node.js version: v14.18.0
- If TypeScript related: TypeScript version: typescript@4.6.3
- Database & Version: PostgreSQL 10
- Connector library & Version: "pg": "^8.7.3"

## Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
- [x] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by ephys** [comment]
I would guess what's happening is that `init` is incorrectly mutating the input instead of making a copy of it

**Comment by danielmnetto** [comment]
I don't think you will need that `await` keyword on the sync command, it does same thing without it. 

Also, try to force Sequelize to recreate your table by setting `{ force: true }` into your `sequelize.sync()` as parameter. Like this:

`sequelize.sync({ force: true })`

After that, remove it so your Node doesn't have to recreate your tables when it reloads and runs this code again.

**Comment by ephys** [comment]
You do need to await `sequelize.sync`, it's going to execute a series of db calls and, if not awaited, the rest of your code could execute before your tables are in the right format

Force wouldn't help with this bug either unfortunately, the test fails on the sscce which is always ran against an empty database
